app/models/user.rb

```diff
@@ -76,6 +83,18 @@ def remember_me
     true
   end
 
+  def suspended?
+    !!self.suspension
+  end
+
+  def can_unsuspend?
+    self.received_conversations.includes(:messages).all? do |convo| 
+      time_difference = (Time.now - convo.created_at)/3600
+      has_responded = convo.messages.any?{|msg| msg.user_id == self.id}
```

Comment:
I think we need to clarify this piece with Alex since right now, can_unsuspend? method will flip to true if a user has responded to any message from a client regardless if there is a client waiting for a response still. The question to ask is if there are multiple clients waiting for a response and the Volunteer responded to some but not others in a 48 hour period, will their account get suspended? Otherwise, this will be true if there is any response in any conversation and there can be a client waiting for a response beyond the 48 hours.

Reply:
Hi Koy, the way can_unsuspend? is currently implemented checks if every conversation for a given volunteer has at least one volunteer response. If any clients have not received responses after 48 hours the volunteer cannot unsuspend their account. 

We didn't want to suspend based on later communications (e.g. if a client were to message "thanks", we don't want to deactivate an account because the volunteer didn't respond to that), so we look at each conversation that is at least two days old and ensure at least some activity on the part of the volunteer.

Let me know if you have any other questions or are still a bit iffy on if this is accomplishing exactly what we're going for!

Reply:
Hi Eric. Thanks for clarifying. Although I still believe there is some flaw to the flow. Not on your end but more UX and needs clarity from Alex. I believe the example isn't for later communications but for multiple clients reaching out and waiting for a response. Example, say a Volunteer gets 3 client messages. Client 1, 2, 3. Each time a client reaches out and creates the first new conversation, a delayed job is created for the Volunteer. Volunteer responds to Client 1 but not 2 or 3. can_unsuspend flips to true.  Let's say 48+ hours passed and their account gets suspended again for not replying to Client 2 or 3. I'm not sure this is a good UX experience to constantly get notification that my account has been suspended. 🤔

Reply:
I see exactly what you mean! Let me know what you come up with, I'd be happy to be involved in the conversation if it helps (I'm sure Jon will want to know what's going on on this front as well). I'll try to have the rest of the comments from you and Alex incorporated come Monday (stylistic changes, dropping the timeouts table).